pkg/policy/l4.go

```diff
@@ -1138,32 +1138,44 @@ func (l4 L4PolicyMap) containsAllL3L4(labels labels.LabelArray, ports []*models.
 	for _, l4Ctx := range ports {
 		portStr := l4Ctx.Name
 		if !iana.IsSvcName(portStr) {
-			portStr = fmt.Sprintf("%d", l4Ctx.Port)
+			portStr = strconv.FormatUint(uint64(l4Ctx.Port), 10)
 		}
 		lwrProtocol := l4Ctx.Protocol
 		var isUDPDeny, isTCPDeny, isSCTPDeny bool
 		switch lwrProtocol {
 		case "", models.PortProtocolANY:
-			tcpPort := fmt.Sprintf("%s/TCP", portStr)
+			tcpPort := portStr + "/TCP"
 			tcpFilter, tcpmatch := l4[tcpPort]
 			if tcpmatch {
 				tcpmatch, isTCPDeny = tcpFilter.matchesLabels(labels)
+				if isTCPDeny {
+					return api.Denied
+				}
```

Comment:
This change is a subtle change in behavior. I'd prefer if it was left out of this PR and done in a separate one, in the case it somehow causes a regression.

Reply:
makes sense, I'll revert this out of this PR and submit separately if desired